### ISSUE TITLE
collaborative editing with y.js

### ISSUE THREAD

**Issue by micahscopes** [Jermolene/TiddlyWiki5#2991]
Hey, I'm new to tiddlywiki development.  I'd like to implement a basic collaborative editing plugin using [y.js](https://github.com/y-js/yjs).

Right now, my basic idea is to have y-js create WebRTC channels from tiddler names, and to attach to them upon editing.  This way, if you open a tiddler to edit while someone else is also editing it, their realtime changes should display.

I hadn't really thought about authentication or identity stuff for this plugin, other than keeping it simple and basic, like requiring a shared password to edit/save, or letting users give show display name so that people can see who's participating.

Originally, I wanted to use together.js, but I saw that y.js works over WebRTC.  I like this better.  Even though together.js has some really nice features, I think WebRTC, and the potential for using this without a server, is more important to me.

So my first question is: what's a clean way to hook into the editor?  Are there any good example plugins I could mimic?

**Comment by micahscopes** [comment]
Hmm... I'm realizing I might be able to do this just by watching the DOM for new editors.

**Comment by micahscopes** [comment]
Okay.  So I'm realizing that probably the best way of handling this is to make a custom editor engine.

**Comment by CodaCodr** [comment]
What about (new) http://prosemirror.net/
?

**Comment by micahscopes** [comment]
@CodaCodr very cool, but it's centralized!

**Comment by Arlen22** [comment]
Centralized? Isn't it in-browser?

**Comment by pmario** [comment]
> I think WebRTC, and the potential for using this without a server, is more important to me.

WebRTC still needs signalling servers to establish a connection between 2 browsers.

**Comment by micahscopes** [comment]
I agree that you'd use a server for WebRTC, but it wouldn't be necessary (
https://github.com/cjb/serverless-webrtc).

More importantly, a signalling server just gets the peers in connection
with each other.  It's doesn't act as a central hub for updates.
